### PR TITLE
[Kotlin Migration] BasicTextFieldController

### PR DESCRIPTION
**Pull request by Akshay0701**

### CODE REVIEW

File: AnkiDroid/src/main/java/com/ichi2/anki/multimediacard/fields/BasicTextFieldController.kt [hunk 1]
```diff
@@ -17,370 +17,305 @@
...
-
-    // Additional activities are started to perform translation/pronunciation search and
-    // so on, here are their request codes, to differentiate, when they return.
-    private static final int REQUEST_CODE_TRANSLATE_GLOSBE = 101;
-    private static final int REQUEST_CODE_PRONOUNCIATION = 102;
-    private static final int REQUEST_CODE_TRANSLATE_COLORDICT = 103;
-
-    private EditText mEditText;
+class BasicTextFieldController : FieldControllerBase(), IFieldController, DialogInterface.OnClickListener {
+    private var mEditText: EditText? = null
```
Comment (david-allison-1) [inline AnkiDroid/src/main/java/com/ichi2/anki/multimediacard/fields/BasicTextFieldController.kt:50]: KotlinCleanup: `lateinit`

File: AnkiDroid/src/main/java/com/ichi2/anki/multimediacard/fields/BasicTextFieldController.kt [hunk 2]
```diff
@@ -17,370 +17,305 @@
...
+                    continue
                 }
-
-                if (curField.getText() == null) {
-                    continue;
-                }
-
-                if (curField.getText().length() == 0) {
-                    continue;
+                if (curField.text.isEmpty()) {
```
Comment (david-allison-1) [inline AnkiDroid/src/main/java/com/ichi2/anki/multimediacard/fields/BasicTextFieldController.kt:170]: Hmm... might be possible to add a null assertion inside `isEmpty()`, so we don't need `text!!` later on: worth investigating

**Review by david-allison-1 — CHANGES_REQUESTED** [review]
Seems reasonable, but to be honest, I'm not familiar with this screen (some functionality is likely broken, others would be best moved to mani's addon PR).

Could you check this one manually to ensure that things aren't more broken

**Comment by Akshay0701** [comment]
okay, will test this manually

File: AnkiDroid/src/main/java/com/ichi2/anki/multimediacard/fields/BasicTextFieldController.kt (hunk 2 above)
Comment (Akshay0701) [reply AnkiDroid/src/main/java/com/ichi2/anki/multimediacard/fields/BasicTextFieldController.kt:170]: yeah there is one method which check both things `isNullorEmpty()`, I just used that here

but even after using this method it still giving the same warning "use !! to text..."
Capture

**Comment by Akshay0701** [comment]
tested this in my phone and emulator, tried almost every features, no crash or aberrant behavior so far...

File: AnkiDroid/src/main/java/com/ichi2/anki/multimediacard/fields/BasicTextFieldController.kt (hunk 2 above)
Comment (ShridharGoel) [reply AnkiDroid/src/main/java/com/ichi2/anki/multimediacard/fields/BasicTextFieldController.kt]: Check this: https://github.com/ankidroid/Anki-Android/pull/10111#discussion_r795241044

File: AnkiDroid/src/main/java/com/ichi2/anki/multimediacard/fields/BasicTextFieldController.kt [hunk 3]
```diff
@@ -0,0 +1,323 @@
...
+import com.ichi2.ui.FixedEditText
+import com.ichi2.ui.FixedTextView
+import com.ichi2.utils.KotlinCleanup
+import timber.log.Timber
+import java.io.File
+import java.util.*
+
+/**
+ * One of the most powerful controllers - creates UI and works with the field of textual type.
+ * <p>
```
[the remark this answers is not in the document]
Comment (Akshay0701) [reply AnkiDroid/src/main/java/com/ichi2/anki/multimediacard/fields/BasicTextFieldController.kt:47]: I guess it's to separate paragraphs of the Javadoc output...

**Review by david-allison-1 — APPROVED** [review]
LGTM

**Review by mikehardy — APPROVED** [review]

**Pull request merged**